gsad/src/gsad_gmp.c

```diff
@@ -1255,11 +1256,27 @@ get_many (gvm_connection_t *connection, const char *type,
             }
           else if (str_equal (type, "result"))
             {
+              gchar *report_id_filter;
+              and_report_id = params_value (params, "_report_id");
+
+              if (and_report_id == NULL || str_equal (and_report_id, ""))
```

Comment:
It may be a good idea to add a check to see if `_and_report_id` is invalid or just missing.

Reply:
I've added a check for the `_and_report_id` to the get_results function because I feared to leave a memory leak if this check gets added to get_many.